backend/layers/processing/dataset_metadata_update.py

```diff
@@ -56,6 +58,37 @@ def __init__(self, artifact_bucket: str, datasets_bucket: str) -> None:
         self.artifact_bucket = artifact_bucket
         self.datasets_bucket = datasets_bucket
 
+    def update_raw_h5ad(
+        self,
+        raw_h5ad_uri: str,
+        new_key_prefix: str,
+        new_dataset_version_id: DatasetVersionId,
+        metadata_update: DatasetArtifactMetadataUpdate,
+    ):
+        raw_h5ad_filename = self.download_from_source_uri(
+            source_uri=raw_h5ad_uri,
+            local_path=CorporaConstants.ORIGINAL_H5AD_ARTIFACT_FILENAME,
+        )
+
+        adata = scanpy.read_h5ad(raw_h5ad_filename)
+        for key, val in metadata_update.as_dict_without_none_values().items():
+            if key in adata.uns:
+                adata.uns[key] = val
+
+        adata.write(raw_h5ad_filename, compression="gzip")
+
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.UPLOAD, DatasetUploadStatus.UPLOADING)
+        self.create_artifact(
+            raw_h5ad_filename,
+            DatasetArtifactType.RAW_H5AD,
+            new_key_prefix,
+            new_dataset_version_id,
+            self.artifact_bucket,
+            DatasetStatusKey.H5AD,
+        )
+        self.update_processing_status(new_dataset_version_id, DatasetStatusKey.UPLOAD, DatasetUploadStatus.UPLOADED)
+        os.remove(raw_h5ad_filename)
```

Comment:
I'd add this to a try: finally: block to make sure it happens. A Context manager would also work.